### ISSUE TITLE
Attachments support

### ISSUE THREAD

**Issue by pik4ez** [frenata/gomua#3]
I use mutt and it has very bad set of tools to manage attachments.

Maybe, there's some way to show attachments in place when you read a message. Maybe, some kind of integration with existing tools like feh for images or mplayer for audio and videos. Of course, it should be optional and you should have ability to pass attachments to external app via native OS mechanism.

Good attachments support is a potential killer-feature that can make Gomua my MUA of choice.

**Comment by gabeguz** [comment]
I know that mutt will use a .mailcap file if it's availalbe to decide what to do with different attachments. Here's an example one: 

```
image/jpeg; feh --auto-zoom --scale-down %s
image/jpg; feh --auto-zoom --scale-down %s
image/png; feh --auto-zoom --scale-down %s
text/html; lynx -assume_charset=%{charset} -dump %s; copiousoutput
application/vnd.ms-excel; soffice %s
application/pdf; zathura %s
```

And a link to the RFC - 
https://tools.ietf.org/html/rfc1524

Not sure if we'd want to do it like this, but there is an existing mechanism for it already.

**Comment by pik4ez** [comment]
I don't say mutt doesn't allow to work with attachments. But it's attachmets management is far from usability heaven.

You should open a list of attachments, traverse the list and open each of them in some external app. Another thing to complain is that you see a lot of thrash in this list. HTML letters can contain text version as attachment, and all the images that is a part of HTML version become a part of attachments list (skype and other icons from signature section, for example).

It would be great if you found any way to show reasonable attachments list or to show attachments near the text.

I didn't really made any research on this, so I'm sorry if I talk about something impossible.

**Comment by gabeguz** [comment]
@pik4ez -- it's definitely a good idea, I wasn't knocking it, just adding some additional information based on how we could start looking at the issue.  I agree, I'm not totally a fan of how mutt handles attachments.     

Part of it is just how MIME messages work that leads to the problem... but we'll see what we can do.

Thanks!

**Comment by frenata** [comment]
Yeah this is a cool idea that would be great to implement, it's just a question of how.
